### DISCUSSION TITLE
Could I have an example?

### DISCUSSION THREAD

**Discussion by dantedantas** [AlexaCRM/dynamics-webapi-toolkit#69] [Q&A] [answered]
Hello,

please, maybe do you have a fully example on how to connect, select records with FetchXML (eg.: incidents) and list the result?
It will be very userful to me and will save me a lot of time.

PS.: I want to try to migrate my APP developed with the php-crm-toolkit to the dynamics-webapi-toolkit.

Thank you for your support.
Best regards,

Dante

**Comment by georged** [comment, marked as answer]
@dantedantas 

we do have [the tutorial](https://github.com/AlexaCRM/dynamics-webapi-toolkit/wiki/Tutorial). I recommend you check it out because we do cover the topics you mentioned.

If you already have an app built using [php-crm-toolkit](https://github.com/alexacrm/php-crm-toolkit/) then most of the code should make it across virtually unchanged. The only difference is how you connect. Why don't you publish your code where you select the records and list the results and we'll help you to convert it to dynamics-webapi-toolkit.

Cheers
George

**Comment by dantedantas** [comment]
Hi Georg,
Please, could you check the code below and tell me is is everything okay with the code?
Thank you!

--

```
namespace AlexaCRM\WebAPI;

use Monolog\Logger;
use Monolog\Handler\StreamHandler;
use Monolog\Handler\FirePHPHandler;

error_reporting(E_ALL & ~E_NOTICE);
$fileName = date("Ymd");
@ini_set( 'error_log', 'home_'.$fileName.'.log' );
@ini_set( 'ignore_repeated_source', true );    
@ini_set( 'ignore_repeated_errors', true );
@ini_set( 'display_errors', true);
@ini_set( 'log_errors', true );
@ini_set( 'error_reporting', E_ALL );

require_once 'vendor/autoload.php';

$settings = new \AlexaCRM\WebAPI\OData\OnlineSettings();
$settings->instanceURI = 'https://xxxxxx.crm4.dynamics.com';
$settings->applicationID = 'dxxx-3xx5-41xx-bxxc-3xxxxx';;
$settings->applicationSecret = 'Lxxxxxxxxxxxxxxxxxxxxxxxxxxx';

$log = new Logger('WebAPI_2');
$log->pushHandler(new StreamHandler('C:\Tools\xampp\htdocs\WebApi2\myLogWebAPI.log', Logger::DEBUG ));
$log->pushHandler(new FirePHPHandler());

$settings->setLogger( $log );
$middleware = new \AlexaCRM\WebAPI\OData\OnlineAuthMiddleware( $settings );
$odataClient = new \AlexaCRM\WebAPI\OData\Client( $settings, $middleware ); 
$client = new \AlexaCRM\WebAPI\Client( $odataClient );

$fetchXML = <<<FETCHXML
<fetch mapping="logical" count="10" paging-cookie="%s" page="%d">
    <entity name="contact">
        <attribute name="fullname"/>
        <attribute name="emailaddress1"/>
    </entity>
</fetch>
FETCHXML;

$page = 1;
$query = new \AlexaCRM\Xrm\Query\FetchExpression( sprintf( $fetchXML, '', $page ) );

$result = $client->RetrieveMultiple( $query );

foreach ( $result->Entities as $contact ) {
    printf( "%s <%s>\n", $contact['fullname'], $contact['emailaddress1'] );
}

while ( $result->MoreRecords ) {
    $query->Query = sprintf( $fetchXML, htmlentities( $result->PagingCookie, ENT_COMPAT | ENT_XML1 ), ++$page );
    $result = $client->RetrieveMultiple( $query );
    foreach ( $result->Entities as $contact ) {
        printf( "%s <%s>\n", $contact['fullname'], $contact['emailaddress1'] );
    }
}
```

**Comment by georged** [comment]
@dantedantas 
I'm afraid generic code reviews is not something we do as part of our work in open source space so unless you're experiencing any particular issue with the toolkit, we won't be able to help.
